Zend/tests/readonly_props/readonly_clone_success2.phpt

```diff
@@ -0,0 +1,74 @@
+--TEST--
+Test that __clone() can write to readonly properties
+--FILE--
+<?php
+
+class Counter
+{
+    private static int $counter = 0;
+
+    public readonly int $count;
+    private readonly int $foo;
+
+    public function __construct()
+    {
+        $this->count = ++self::$counter;
+        $this->foo = 0;
+    }
+
+    public function count(?int $count = null): static
+    {
+        $new = clone $this;
+        $new->count = $count ?? ++self::$counter;
+
+        return $new;
+    }
+
+    public function __clone()
+    {
+        if (is_a(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['class'] ?? '', self::class, true)) {
```

Comment:
This test is a bit more cryptic than it needs to be, can we split this into two tests? One testing if properties can be unset and then reassigned outside `__clone`, and one that assigns it directly in `__clone`.

Reply:
This test comes from Nicolas, but I'm not even sure that it really is needed in its whole. I guess only the unset + reassign part is not completely covered by other tests. I can update Zend/tests/readonly_props/readonly_clone_success3.phpt though.

Reply:
> unset + reassign part is not completely covered by other tests

agree, we can simplify the test for sure